### DISCUSSION TITLE
Final fields

### DISCUSSION THREAD

**Discussion by danielpaukowitsch** [pocketbase/pocketbase#2563] [Q&A] [answered]
Is there a way to mark a field as "`final`" or "`updatable=false`" other than using API rules?
e.g., "owner" (user relation)

Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
No, there is no such option at the moment. The simplest way probably would be to use the `@request.*:isset` field rule modifier:

```js
// disallow submitting new yourField value, but at the same time allow admins to modify it
@request.data.yourField:isset = false
```

**Reply by dschissler** [reply]
@ganigeorgiev Just a thought, a white or black list for fields could be useful.  I'm not sure if it needs both, but the idea seems good.  I'm currently using the `:isset = false` method, but it kind of freaks me the hell out.  I'm always trying to keep my filter rules as small as possible because it seems like having complicated logic could really be a big brittle problem.

**Reply by danielpaukowitsch** [reply]
@ganigeorgiev  Thanks! 

@dschissler Just a second thought, adding a checkbox option, such as 'updatable' or 'final,' to the field settings would be sufficient, rather than adding a white or black list. This would also be easier to implement (from a UI/UX perspective).

**Reply by dschissler** [reply]
@danielpaukowitsch That's basically just a whitelist of blacklist with a UI around it.

**Comment by VittorioParagallo** [comment]
i upvote this feature. 
@request.data.yourField:isset = false works but creates chaos in rules
